messaging/src/main/java/org/axonframework/messaging/timeout/AxonTimeLimitedTask.java

```diff
@@ -0,0 +1,267 @@
+     * {@code warningThreshold} and {@code warningInterval}. Runs the provided task on the current thread after
+     * scheduling a timeout and warnings on another thread.
+     * <p>
+     * If you wish to provide a logger of your own, or your own {@code scheduledExecutorService}, use
+     * {@link #AxonTimeLimitedTask(String, int, int, int, ScheduledExecutorService, Logger)}.
+     *
+     * @param taskName         The task's name to be included in the logging
+     * @param timeout          The timeout in milliseconds
+     * @param warningThreshold The threshold in milliseconds after which a warning is logged. Setting this to a value
+     *                         equal or higher than {@code timeout} will disable warnings.
+     * @param warningInterval  The interval in milliseconds between warnings.
+     */
+    public AxonTimeLimitedTask(String taskName,
+                               int timeout,
+                               int warningThreshold,
+                               int warningInterval) {
+        this(taskName,
+             timeout,
+             warningThreshold,
+             warningInterval,
+             AxonTaskJanitor.INSTANCE,
+             AxonTaskJanitor.LOGGER
+        );
+    }
+
+    /**
+     * Creates a new {@link AxonTimeLimitedTask} for the given {@code task} with the given {@code timeout},
+     * {@code warningThreshold} and {@code warningInterval}. For scheduling, the provided
+     * {@code scheduledExecutorService} will be used. To log warnings and errors, the provided {@code logger} will be
+     * used. Runs the provided task on the current thread after scheduling a timeout and warnings on the provided
+     * {@code scheduledExecutorService}.
+     * <p>
+     * If you do not wish to provide a logger of your own {@code scheduledExecutorService}, use
+     * {@link #AxonTimeLimitedTask(String, int, int, int)}.
+     * <p>
+     *
+     * @param taskName                 The task's name to be included in the logging
+     * @param timeout                  The timeout in milliseconds
+     * @param warningThreshold         The threshold in milliseconds after which a warning is logged. Setting this to a
+     *                                 value equal or higher than {@code timeout} will disable warnings.
+     * @param warningInterval          The interval in milliseconds between warnings.
+     * @param scheduledExecutorService The executor service to schedule the timeout and warnings
+     * @param logger                   The logger to log the warnings and errors
+     */
+    public AxonTimeLimitedTask(String taskName,
+                               int timeout,
+                               int warningThreshold,
+                               int warningInterval,
+                               ScheduledExecutorService scheduledExecutorService,
+                               Logger logger) {
+        this.taskName = taskName;
```

Comment:
I think it's good if we check if the given `taskName` is non-null and non empty. Otherwise, we may potentially log warnings without specifying the task at hand.

Reply:
Fair enough! Adding